src/xtopcom/xbasic/xlru_cache_specialize.h

```diff
@@ -131,30 +133,40 @@ class xtop_lru_cache {
     xtop_lru_cache(size_t const max_size) : max_size_{max_size} {
     }
 
-    void put(const KeyT & key, const ValueT & value) {
+    std::vector<std::pair<KeyT, ValueT>> put(const KeyT & key, const ValueT & value) {
         std::lock_guard<MutexT> lock(mutex_);
 
+        std::vector<std::pair<KeyT, ValueT>> erased_vec;
         auto it = item_map_.find(key);
         if (it != item_map_.end()) {
+#if defined(CACHE_SIZE_STATISTIC) || defined(CACHE_SIZE_STATISTIC_MORE_DETAIL)
+            erased_vec.push_back({key, it->second->second});
+#endif
             item_list_.erase(it->second);
             item_map_.erase(it);
         }
         item_list_.push_front({key, value});
         item_map_.insert({key, item_list_.begin()});
-        clean_with_lock_hold();
+        clean_with_lock_hold(erased_vec);
+        return erased_vec;
```

Comment:
return erased vector will influence performance

Reply:
> return erased vector will influence performance. may add api put_and_return_erased
return a vector is OK, C++ has RVO/NRVO

Reply:
> return erased vector will influence performance. may add api put_and_return_erased

return a vector is OK for C++ due to ROV/NROV

Reply:
ok,fixed.